include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -489,8 +493,7 @@ def TT_BroadcastOp : TT_Op<"broadcast", [Pure,
     let hasVerifier = 1;
 }
 
-// cat is not `pure` because it may reorder elements
-def TT_CatOp : TT_Op<"cat", [NoMemoryEffect,
```

Comment:
I'm confused about making it pure. Isn't the comment saying that the op shouldn't be pure?

Reply:
I think the comment was wrong and meant to say elementwise. It is pure, it just isn't elementwise

Reply:
The comment was out-of-date because `canReorder` was recently removed from the operation.